src/client/v2/algod/getLedgerStateDeltaForTransactionGroup.ts

```diff
@@ -1,17 +1,24 @@
 import JSONRequest from '../jsonrequest.js';
-import { HTTPClient } from '../../client.js';
+import { HTTPClient, HTTPClientResponse } from '../../client.js';
+import { decodeMsgpack } from '../../../encoding/encoding.js';
+import { LedgerStateDelta } from '../../../types/statedelta.js';
 
-export default class GetLedgerStateDeltaForTransactionGroup extends JSONRequest {
+export default class GetLedgerStateDeltaForTransactionGroup extends JSONRequest<LedgerStateDelta> {
   constructor(
     c: HTTPClient,
     private id: string
   ) {
     super(c);
-    this.query = { format: 'json' };
+    this.query = { format: 'msgpack' };
```

Comment:
Noting the change from json here...

Reply:
The main reason for this is that LedgerStateDeltas contain strings which are actually binary data. We can properly decode these from msgpack using the recently added special case binary string schemas. From JSON it's impossible to decode these properly, as the binary strings have already been converted to UTF-8.